modules/dnn/include/opencv2/dnn/dnn.hpp

```diff
@@ -1135,6 +1147,76 @@ CV__DNN_INLINE_NS_BEGIN
                                    const Scalar& mean = Scalar(), bool swapRB=false, bool crop=false,
                                    int ddepth=CV_32F);
 
+    /**
+     * @brief Enum of image processing mode.
+     * @see ImagePParam
+     */
+    enum ImageProcessMode
+    {
+        DNN_PMODE_CENTER = 0,    // !< Default
+        DNN_PMODE_LETTERBOX = 1, // !< Resize image to the desired size while preserving the aspect ratio of original image.
+    };
+
+    /** @brief Image processing parameters.
+     *
+     * It includes all possible image processing operations and corresponding parameters.
+     *
+     * @see blobFromImageParam
+     * @note
+     * The order and usage of `scalefactor`, `size`, `mean`, `swapRB`, `crop` and `ddepth` are consistent
+     * with the function of @ref blobFromImage.
+    */
+    struct CV_EXPORTS_W_SIMPLE ImagePParam
```

Comment:
`ImagePParam` or just `ImageParam`? Any suggestion？

Reply:
IMHO, `ImageParam` is too generic. We may have conflicts in the future.

- `Image2BlobParams` ?

---

BTW, any thoughts about `blob -> tensor` migration? Seems `blob` was widely used in Caffe only.

Reply:
> BTW, any thoughts about blob -> tensor migration? Seems blob was widely used in Caffe only.

I just found there is `Tensor` in `5.x`. Maybe in this PR, we can change the `blob`->`tensor`. Hi @vpisarev, any suggestions?

Reply:
As a result of the discussion with Vadim, the current `tensor` used in `5.x` is also temporary, and this PR still uses `blob`.

Reply:
> `Image2BlobParams`

fixed.